### PR TITLE
Chore main refactor

### PR DESCRIPTION
**Pull request by sfranky**
* Showing runs from 3 functional tests, one for each scheduler; all good.
* SGE now shows again an extra column with queue information, this is where the diff comes from
* Should be ready to merge with little worry

```
Testing sge...
9,25c9,25
< da-p4-04|0000000000001___________|r  |a |
< da-p5-01|000000000000____________|E  |a |
< da-p5-02|000000000000____________|dr |a |
< da-p5-03|000000000000____________|R  |a |
< da-p5-04|000000000000____________|hqw|a |
< da-p6-01|________________________|au |cw|
< da-p6-02|________________________|-  |  |
< da-p6-03|________________________|au |  |
< da-p6-04|________________________|-  |  |
< da-p7-01|________________________|au |  |
< da-p7-02|________________________|E  |  |
< da-p7-03|________________________|-  |  |
< da-p7-04|________________________|-  |  |
< da-p8-01|________________________|E  |  |
< da-p8-02|________________________|E  |  |
< da-p8-03|2_______________________|au |  |
< da-p8-04|________________________|E  |  |
---
> da-p4-04|0000000000001___________|r  |
> da-p5-01|000000000000____________|E  |
> da-p5-02|000000000000____________|dr |
> da-p5-03|000000000000____________|R  |
> da-p5-04|000000000000____________|hqw|
> da-p6-01|________________________|au |
> da-p6-02|________________________|-  |
> da-p6-03|________________________|au |
> da-p6-04|________________________|-  |
> da-p7-01|________________________|au |
> da-p7-02|________________________|E  |
> da-p7-03|________________________|-  |
> da-p7-04|________________________|-  |
> da-p8-01|________________________|E  |
> da-p8-02|________________________|E  |
> da-p8-03|2_______________________|au |
> da-p8-04|________________________|E  |
Testing oar...
Testing pbs...
```

### CODE REVIEW

File: common_module.py [hunk 1]
```diff
@@ -96,23 +150,22 @@ def handle_exception(exc_type, exc_value, exc_traceback):
                   help="Rotate matrices' positioning by 90 degrees")
 parser.add_option("-v", "--verbose", dest="verbose", action="count",
                   help="Increase verbosity (specify multiple times for more)")
-parser.add_option("-W", "--writemethod", dest="write_method", action="store", default="txtyaml",
-                  choices=['txtyaml', 'json'],
-                  help="Set the method used for dumping information, json, yaml, or native python (yaml format)")
+# TODO: dumping to intermediate yaml files has been deprecated.
+# It is now possible, instead, to dump an all-including python structure into a json file (the "document")
```
Comment (fgeorgatos) [inline common_module.py]: how is that possible? just asking...

File: qtop.py [hunk 2]
```diff
@@ -651,7 +604,7 @@ def calc_general_multiline_attr(cluster_dict, part_name, yaml_key):  # NEW
         multiline_map[attr_line] = ''.join(multiline_map[attr_line])
         if line == user_max_len:
             break
-
+    # import wdb; wdb.set_trace()  # scheduler in part_options['systems']
```
Comment (fgeorgatos) [inline qtop.py]: TODO: ??

File: qtop.py [hunk 3]
```diff
@@ -1100,8 +1034,17 @@ def execute_shell_batch_commands(batch_system_commands, filenames, _file):
...
+    os.rename(tempname, filenames[_file])
+
+    return filenames[_file]
 
-    logging.debug('File state after subprocess call: %(fin)s' % {"fin": fin})
+
+# with tempfile.NamedTemporaryFile('w', dir=os.path.dirname(filename), delete=False) as fin:
+#     fin.write(model.output())
+#     tempname = fin.name
+# os.rename(tempname, filename)
```
Comment (fgeorgatos) [inline qtop.py]: TODO: ??

File: qtop.py [hunk 4]
```diff
@@ -1523,20 +1436,22 @@ def display_matrix(self, workernodes_occupancy):
...
+                    self.print_core_lines,
                     (core_user_map, print_char_start, print_char_stop, transposed_matrices, pattern_of_id),
                     {'attrs': None}
                 ),
         }
 
         # custom part
-        for yaml_key, part_name in get_yaml_key_part('workernodes_matrix'):
+        for yaml_key, part_name, systems in get_yaml_key_part(outermost_key='workernodes_matrix'):
+            # import wdb; wdb.set_trace()
```
Comment (fgeorgatos) [inline qtop.py]: ??

File: qtop.py [hunk 5]
```diff
@@ -1606,14 +1524,92 @@ def join_prints(self, *args, **kwargs):
...
+            node_str_width = len(str(highest_wn))  # 4 for thousands of nodes, nr of horizontal lines to be displayed
+
+            for node_nr in range(1, node_str_width + 1):
+                d[str(node_nr)] = "".join(wn_vert_labels[str(node_nr)])
+            end_labels_iter = iter(end_labels[str(node_str_width)])
+            print_wnid_lines(d, start, stop, end_labels_iter, transposed_matrices,
+                             color_func=self.color_plainly, args=('White', 'Gray_L', start > 0))
+            # start > 0 is just a test for a possible future condition
+
+        elif NAMED_WNS or options.FORCE_NAMES:  # names (e.g. fruits) instead of numbered WNs
```
Comment (fgeorgatos) [inline qtop.py]: very few people will understand what "fruits" is; say arbitrary strings, instead

File: common_module.py (hunk 1 above)
Comment (sfranky) [inline common_module.py]: see line 1251 in qtop.py. Briefly, a named tuple is created, combining the data from all scheduler files, and then saved in json format.

File: serialiser.py [hunk 6]
```diff
@@ -191,47 +34,54 @@ def _process_line(self, re_search, line, re_match_positions):
             qstat_values[key] = value
         return qstat_values
 
+    def anonymize_func(self):
+        """
+        creates and returns an _anonymize_func object (closure)
+        Anonymisation can be used by the user for providing feedback to the developers.
```
Comment (fgeorgatos) [inline serialiser.py]: ", without leaking cluster's private data."

**Pull request merged**